### DISCUSSION TITLE
스터디 진행시 다음 페이지로 넘어가는 주도권을 누가 가질 것인가

### DISCUSSION THREAD

**Discussion by aak2075** [woowacourse-teams/2023-haru-study#18] [아이디어]
1. 시간 정보만 주고 다음으로 넘기는 것은 사용자의 선택으로 넘긴다.
2. 시간이 되면 강제로 다음 페이지로 넘긴다.

**Comment by aak2075** [comment]
1. 우리는 스터디를 이렇게 하면 좋아요~ 하고 가이드를 하는 것이기 때문에 시간이 넘어갔다고 우리가 강제로 페이지를 넘기면 사용자는 불편할 것. 이런 시간은 스터디장이 설정할 것이고 스터디 원은 처음 만난 스터디장이 설정한 시간에 따라가야 하기 때문에 시간이 부족한 사람은 유동적으로 기다려 준다거나 하는 것이 필요하다고 생각한다.

**Comment by woo-jk** [comment]
1번 - 굳이 강제로 넘기는게 사용자 경험 측면이 좋을지 잘 모르겠다. 그냥 타이머가 끝나면 알림만 주고 다음 페이즈로 넘어가는 타이밍은 스터디장이 조절하는게 스터디 흐름 면에서 더 매끄러울 것 같다.

**Reply by woosung1223** [reply]
반대로 한 사람이 늦어져서, 다른 스터디원들이 기다려야 하는 상황이 반복되는 경우에도 UX가 떨어지는 게 아닌가.. 라고 볼 수도 있을 것 같아요. 어떻게 생각하시는지 궁금합니다!

**Reply by aak2075** [reply]
그것도 사용자에게 맡길 수 있지 않을까요?
다른 사람이 늦는다면 스터디 장이 다른 사람은 넘어가고 늦는 분은 다 쓰고 화면을 넘겨 주세요~ 할 수 있을 것 같아요

**Reply by woo-jk** [reply]
제 생각에는 한 사람이 늦어지는 거랑 시간 도달시 자동으로 넘어가는거랑 어떤 상관인지 잘 모르겠어요. 스터디 장이 늦는다는 소리일까요?

**Reply by woosung1223** [reply]
그러면 버퍼처럼 전부 기다리는 방식이 아닌 먼저 한 사람은 넘어가는 방식을 의미하시는거군용
그런 경우에는 마지막에 취합해서 보여주는 과정에 문제가 생기지 않을까요?!

**Comment by woosung1223** [comment]
> 2. 시간이 되면 강제로 다음 페이지로 넘긴다.

장기적으로 보았을 때, 스터디장에게 권한을 줘야 한다는 것에는 동의하지만(UX의 측면에서)
그것이 굳이 2차 데모여야 하는가? 에는 동의가 되지는 않는 것 같아요. 

`그리고 과연 스터디장이 권한을 원할지?` 에 대해서도 다시 생각해 볼 필요가 있다고 생각합니다.
지금은 `스터디장이 권한을 가지길 원할 것이다` 라고 추측하고 있지만, 자동으로 넘어가는 서비스를 원하는 사용자가 더 많을수도 있지 않을까요?

유저 반응에 따라 결정해도 늦지 않다고 생각합니다!

**Reply by aak2075** [reply]
스터디장이 진행을 맡기 싫을 수 있겠네요.
일단 편한대로 구현하고 유저 반응에 따라 웹소켓으로 구현해서 고도화하는 것도 좋아 보입니다.

**Comment by yeopto** [comment]
1. 방장만 권한을 가지고 방장이 다음으로 넘기면 뭐 줌에서나 다음 버튼 눌러주세요라고 말하지 않을까 생각한다. 그때 사람들이 누르면 되지 않을까?

**Reply by yeopto** [reply]
gg 선언 뭐든 좋음

**Reply by aak2075** [reply]
애정이 없는건가요😭

**Comment by jaehee329** [comment]
2. 

강제로 넘기면 스터디장이 일일이 다 쓰셨나요? 물어보고 이제 넘어갈게요~ 안 해도 된다는 것?
또한 서버 내에서 제출 타이밍마다 다 제출됐는지 확인하고 알아서 스터디 상태를 넘겨줄 수 있다.

제출 후 일정 시간 유예를 주면 서버적으로도 동기화 할 시간이 충분하지 않을까? (제출(POST) 후 10초정도 뒤 일괄 조회(GET)

스터디원이 하나하나 다 했는지 확인하는 스터디 과정에서의 불편함 vs 자동으로 스터디가 계속 진행돼서 가끔 입력 다 못했는데 넘어가는 등의 불편함
을 비교하면 될 듯. 나는 뽀모도로라는 포멧이 원래 시간에 맞춰 진행하는 컨셉인데, 사이클 단위로 다음 단계 시작을 수동으로 할 순 있어도 매번 다음 단계를 수동으로 해야 할까 싶다.

**Reply by woo-jk** [reply]
이제 넘어 갈게요~ 하고 넘어가는게 스터디원들이 흐름을 이해하기에 더 좋지 않을까요? 
스터디원이 목표 제출하고 다른 일 하고 있었는데 아무 말도 없이 다음 단계 넘어가면 그게 더 불편할 것 같아용

**Reply by jaehee329** [reply]
짧은 시간을 콤팩트하게 쓰는 게 우리 템플릿의 목표인데
집중 중인 스터디원들이 자동으로 흐름을 넘어가는 것에 불편함을 많이 느낄까요..?

**Reply by jaehee329** [reply]
추가로 달자면, 연속으로 폼을 입력해야 하는 상황 (ex) 5분간 목표 설정 -> 달성 확률 추정 -> ...) 이 있을 때 이것들을 한 폼에 담아서 보내게 될까요?
맞다면 UI가 어떻게 구성될진 모르겠지만 괜찮고,

아니라면 매 항목마다 방장이 다 썼나 동기화해야 해서 불편함이 많이 커질 듯 해요.

**Comment by aak2075** [comment]
의견이 분분하여 우선 1번의 방식으로 구현하고 설문 등 사용자 피드백을 받고 나서 개선하는 것으로 결정.
